search.go

```diff
@@ -0,0 +1,169 @@
+package opensearchtools
+
+import (
+    "bytes"
+    "context"
+    "encoding/json"
+    "net/http"
+
+    "github.com/opensearch-project/opensearch-go/v2"
+    "github.com/opensearch-project/opensearch-go/v2/opensearchapi"
+
+    "github.com/CrowdStrike/opensearchtools/search"
+)
+
+// SearchRequest wraps the functionality of [opensearchapi.SearchRequest] by supporting request body creation.
+// A simple term query search as an example:
+//
+//	req := NewSearchRequest()
+//	req.SetIndices("example_index")
+//	req.SetQuery(search.NewTermQuery("field", "basic")
+//	results, err := req.Do(context.Background(), client)
+type SearchRequest struct {
+    Query search.Query
+    Index []string
+    Size  *int
```

Comment:
Why does this need to be a pointer?  Consumers will have to declare a separate variable and set this field to point to it, and that feels gross.

Reply:
Consumers can pass in a normal int in the `SetSize` method, and then take to the pointer of that value. So they don't have to declare the separate variable.

I did it to control the default value. A normal int defaults to 0, so it'd be hard to tell the difference between a default and a value to ignore. I suppose I could have the constructor set it to -1 instead. That's probably a little cleaner...

Reply:
Yes, I do it all the time on manual queries. Whenever I manually do an aggregation request, I don't care about the search results so I always set it to 0 to hide them. 

This is maybe not as applicable for an client wrapper. But it certainly is an optimization choice to reduce the data in flight.